test/jdk/java/math/BigInteger/MutableBigIntegerShiftTests.java

```diff
@@ -0,0 +1,194 @@
+                        random.nextInt(1, 17) },
+                // intLen != 0 && n > leadingZeros && nBits <= leadingZeros && value.length < newLen && nBits == 0
+                { new MutableBigIntegerBox(new int[] { (int) random.nextLong(1L, 1L << 32) }),
+                        32 },
+                // intLen != 0 && n > leadingZeros && nBits <= leadingZeros && value.length < newLen && nBits != 0
+                { new MutableBigIntegerBox(new int[] { (int) random.nextLong(1L, 1L << 16) }),
+                        32 + random.nextInt(1, 17) },
+                // intLen != 0 && n > leadingZeros && nBits <= leadingZeros && value.length >= newLen && nBits == 0
+                // && newOffset != offset
+                { new MutableBigIntegerBox(new int[] { random.nextInt(), (int) random.nextLong(1L, 1L << 32) }, 1, 1),
+                        32 },
+                // intLen != 0 && n > leadingZeros && nBits <= leadingZeros && value.length >= newLen && nBits == 0
+                // && newOffset == offset
+                { new MutableBigIntegerBox(new int[] { (int) random.nextLong(1L, 1L << 32), random.nextInt() }, 0, 1),
+                        32 },
+                // intLen != 0 && n > leadingZeros && nBits <= leadingZeros && value.length >= newLen && nBits != 0
+                // && newOffset != offset
+                { new MutableBigIntegerBox(new int[] { random.nextInt(), (int) random.nextLong(1L, 1L << 16) }, 1, 1),
+                        32 + random.nextInt(1, 17) },
+                // intLen != 0 && n > leadingZeros && nBits <= leadingZeros && value.length >= newLen && nBits != 0
+                // && newOffset == offset
+                { new MutableBigIntegerBox(new int[] { (int) random.nextLong(1L, 1L << 16), random.nextInt() }, 0, 1),
+                        32 + random.nextInt(1, 17) },
+                // intLen != 0 && n > leadingZeros && nBits > leadingZeros && value.length < newLen
+                { new MutableBigIntegerBox(new int[] { (int) random.nextLong(1L << 15, 1L << 32) }),
+                        random.nextInt(17, 32) },
+                // intLen != 0 && n > leadingZeros && nBits > leadingZeros && value.length >= newLen && newOffset != offset
+                { new MutableBigIntegerBox(new int[] { random.nextInt(), (int) random.nextLong(1L << 15, 1L << 32) }, 1, 1),
+                        random.nextInt(17, 32) },
+                // intLen != 0 && n > leadingZeros && nBits > leadingZeros && value.length >= newLen && newOffset == offset
+                { new MutableBigIntegerBox(new int[] { (int) random.nextLong(1L << 15, 1L << 32), random.nextInt() }, 0, 1),
+                        random.nextInt(17, 32) },
+        };
+
+        for (int i = 0; i < cases.length; i++) {
+            MutableBigIntegerBox x = (MutableBigIntegerBox) cases[i][0];
+            int n = (int) cases[i][1];
+            leftShiftAssertions(x, n);
+        }
+    }
```

Comment:
I think this can be made `@ParameterizedTest` by refactoring `cases` into a method?

Otherwise looks good.

Reply:
This doesn't run.

You should use JUnit's `Arguments` type and `arguments` method to build the test cases with [multiple parameters](https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests-sources-MethodSource), and change the header of the parameterized test to reflect them.